Benqi-Agents/DelegationMonitor/src/agent.ts

```diff
@@ -0,0 +1,86 @@
+import {
+  Finding,
+  TransactionEvent,
+  FindingSeverity,
+  FindingType,
+  getEthersProvider,
+  ethers,
+  HandleTransaction,
+} from "forta-agent";
+import { BigNumber, providers, utils } from "ethers";
+import LRU from "lru-cache";
+import util from "./utils";
+
+const QI_CONTRACT: string = "0xf5841E5df027ed086A749Ecc33BFE7e0Df02B06F";
+const AMOUNT_THRESHOLD: BigNumber = utils.parseEther("10000000.0");
+export const createFinding = (
+  delegator: string,
+  fromDelegate: string,
+  toDelegate: string,
+  balance: BigNumber
+): Finding => {
+  return Finding.fromObject({
+    name: "Large votes delegation detected",
+    description: "Detect user with a huge balance delegating their votes",
+    alertId: "BENQI-2",
+    severity: FindingSeverity.Info,
+    type: FindingType.Info,
+    protocol: "Benqi Finance",
+    metadata: {
+      delegator: delegator.toLowerCase(),
+      fromDelegate: fromDelegate.toLowerCase(),
+      toDelegate: toDelegate.toLowerCase(),
+      balance: balance.toString(),
+    },
+  });
+};
+
+export function provideHandleTransaction(
+  amountThreshold: BigNumber,
+  QiToken: string,
+  provider: providers.Provider
+): HandleTransaction {
+  const cache: LRU<string, BigNumber> = new LRU<string, BigNumber>({
+    max: 10000,
+  });
+
+  const BenqiContract = new ethers.Contract(QiToken, util.BALANCE_OF_FUNCTION, provider);
+  return async (txEvent: TransactionEvent) => {
+    const findings: Finding[] = [];
+    const delegateChangedEvents = txEvent.filterLog(util.DELEGATE_CHANGED_EVENT, QiToken);
+    await Promise.all(
+      delegateChangedEvents.map(async (event) => {
+        const delegator: string = event.args.delegator;
+
+        const blockNumber: number = txEvent.blockNumber;
+        const key: string = `${delegator}-${blockNumber}`;
+
+        let balanceOfDelegator: BigNumber;
+        if (cache.has(key)) {
+          balanceOfDelegator = cache.get(key) as BigNumber;
+        } else {
+          balanceOfDelegator = await BenqiContract.balanceOf(delegator, {
+            blockTag: txEvent.blockNumber,
+          });
+          cache.set(key, balanceOfDelegator);
+        }
+
+        if (balanceOfDelegator.gte(amountThreshold)) {
+          const newFinding: Finding = createFinding(
+            event.args.delegator,
+            event.args.fromDelegate,
+            event.args.toDelegate,
+            balanceOfDelegator
+          );
+          findings.push(newFinding);
+        }
+      })
+    );
+
+    return findings;
+  };
+}
+
+export default {
+  handleTransaction: provideHandleTransaction(AMOUNT_THRESHOLD, QI_CONTRACT, getEthersProvider()),
+};
```

Comment:
Please check the code formatting, some examples/ recommendations are:
- in `agent.ts`: I would recommend an empty line between L15-L16, the empty line 54 should be removed

- in `agent.spec.ts`: 
   - Would be better for readability to leave an empty line between test cases, you're doing that for some and none for others. 
  - Empty line 88 should be removed. Moreover, I'd recommend always adding an empty line before the code checking the results ( usually the last two lines of the test case)